File: June LeetCoding Challenge 2021/Week Four/Candy.py
class Solution:
    def candy(self, ratings) -> int:
        last = -1
        level = 0
        run = 1
        total = 0
        for rating in ratings:
            if rating > last:
                if run == 1:
                    level += 1
                else:
                    level = 2
                run = 1
                total += level
            elif rating == last:
                run = 1
                level = 1
                total += 1
            else:
                total += run
                run += 1
                if (level < run):
                    total += 1
            last = rating
        return total

File: June LeetCoding Challenge 2021/Week Four/Candy.cpp
#include<iostream>
#include<vector>
using namespace std;

class Solution {
    public:
        int candy(vector<int>& ratings) {
            int n=ratings.size();
            vector<int> left(n,1);
            vector<int> right(n,1);
            int i;
            for(i=1;i<n;i++)
            {
                if(ratings[i]>ratings[i-1])
                    left[i]=left[i-1]+1;
            }
            for(i=n-2;i>=0;i--)
            {
                if(ratings[i]>ratings[i+1])
                    right[i]=right[i+1]+1;
            }
            int total=0;
            for(i=0;i<n;i++)
            {
                total=total+max(left[i],right[i]);
            }
            return total;
        }
};